jujugui/static/gui/src/app/app.js

```diff
@@ -1011,11 +1011,82 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Get user name from cookie.
+
+      @method _getUsernameFromCookie
+    */
+    _getUsernameFromCookie: function() {
+      // XXX jcsackett 2016-01-08: We should be using the whoami endpoint on
+      // charmstore to get the username of a logged in jem user rather than
+      // parsing the cookie. Consider this all a bad hack for demo and we'll do
+      // something better in the very near future.
+
+      // Don't look for cookies if in sandbox.
+      if (this.get('sandbox')) {
+        return null;
+      }
+      var cookie = document.cookie;
```

Comment:
instead of grabbing the document here...

create a method:
```Javascript
_getCookies: function() {
  document.cookie;
}
```

then this becomes

```Javascript
var cookie = this._getCookies();
```

And in the tests you can then stub out `_getCookies`